src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -36,19 +38,29 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
     protected $dispatcher;
     protected $resolver;
     protected $requestStack;
+    private $argumentResolver;
 
     /**
      * Constructor.
      *
-     * @param EventDispatcherInterface    $dispatcher   An EventDispatcherInterface instance
-     * @param ControllerResolverInterface $resolver     A ControllerResolverInterface instance
-     * @param RequestStack                $requestStack A stack for master/sub requests
+     * @param EventDispatcherInterface    $dispatcher       An EventDispatcherInterface instance
+     * @param ControllerResolverInterface $resolver         A ControllerResolverInterface instance
+     * @param RequestStack                $requestStack     A stack for master/sub requests
+     * @param ArgumentResolverInterface   $argumentResolver An ArgumentResolverInterface instance
      */
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null)
```

Comment:
We indeed don't provide BC for cases where they overwrite the constructor without calling our parent constructor (we generally don't even provide a guarantee that it will work in the current version)